### ISSUE TITLE
oxide test fails when using MKL

### ISSUE THREAD

**Issue by dmbates** [JuliaStats/MixedModels.jl#448]
```julia
oxide: Test Failed at /home/bates/.julia/dev/MixedModels/test/pls.jl:413
  Expression: all(isapprox.(m.θ, [1.689182746, 2.98504262]; atol = 0.0001))
Stacktrace:
 [1] top-level scope at /home/bates/.julia/dev/MixedModels/test/pls.jl:413
 [2] top-level scope at /home/bates/src/julia-1.5.3/share/julia/stdlib/v1.5/Test/src/Test.jl:1115
 [3] top-level scope at /home/bates/.julia/dev/MixedModels/test/pls.jl:412
Test Summary: | Pass  Fail  Total
oxide         |    1     1      2
```
```julia
julia> versioninfo()
Julia Version 1.5.3
Commit 788b2c77c1 (2020-11-09 13:37 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, skylake)

julia> using LinearAlgebra; BLAS.vendor()
:mkl
```

**Comment by palday** [comment]
This doesn't surprise me too much. This is failing at the intercept-only model, right? The slopes model seemed to have a rather poorly defined optimum and the intercept model didn't seem particularly great. Should we just loosen the tolerance for this test since it's mostly about testing a particular nesting structure and associated linear algebra methods? If we find a better test dataset, we can swap it out.

**Comment by dmbates** [comment]
It may be possible to skip the densify step on a rectangular block if the diagonal block on the same row is `Diagonal` or `UniformBlockDiagonal`.  I'll take a look.

I get anxious about a `rankUpdate!` of a `Diagonal` by a dense matrix because the dense matrix must have a specific form (exactly one nonzero in each column) and it would take more time to check for that form than to perform the update.  In some ways it would be better to leave it as a sparse matrix.

**Comment by palday** [comment]
We could delay the call to `densify` and do it columnwise.

**Comment by dmbates** [comment]
> We could delay the call to `densify` and do it columnwise.

I am trying something like that now.

**Comment by palday** [comment]
I am pleasantly *not* surprised. :smile:

**Comment by palday** [comment]
@dmbates Is this closed by #456?

**Comment by dmbates** [comment]
Yes, this can be closed.
